src/Uploadcare/Api.php

```diff
@@ -33,6 +34,13 @@ class Api
      */
     private $api_host = 'api.uploadcare.com';
 
+    /**
+     * AuthenticatedUrl concrete realization
+     *
+     * @var Authenticate\AuthenticatedUrlInterface
+     */
+    public $authenticatedUrl;
```

Comment:
why do we want this public?

Reply:
strategy pattern, we use it in File class to make file->getUrl polymorphic transparent